internal/sysfs/stat_linux.go

```diff
@@ -10,52 +10,33 @@ import (
 	"os"
 	"syscall"
 
-	"github.com/tetratelabs/wazero/internal/fsapi"
 	"github.com/tetratelabs/wazero/internal/platform"
+	"github.com/tetratelabs/wazero/sys"
 )
 
-func lstat(path string) (fsapi.Stat_t, syscall.Errno) {
-	if t, err := os.Lstat(path); err != nil {
-		return fsapi.Stat_t{}, platform.UnwrapOSError(err)
+func lstat(path string) (sys.Stat_t, syscall.Errno) {
+	if info, err := os.Lstat(path); err != nil {
+		return sys.Stat_t{}, platform.UnwrapOSError(err)
 	} else {
-		return statFromFileInfo(t), 0
+		return sys.NewStat_t(info), 0
 	}
 }
 
-func stat(path string) (fsapi.Stat_t, syscall.Errno) {
-	if t, err := os.Stat(path); err != nil {
-		return fsapi.Stat_t{}, platform.UnwrapOSError(err)
+func stat(path string) (sys.Stat_t, syscall.Errno) {
+	if info, err := os.Stat(path); err != nil {
+		return sys.Stat_t{}, platform.UnwrapOSError(err)
 	} else {
-		return statFromFileInfo(t), 0
+		return sys.NewStat_t(info), 0
 	}
 }
 
-func statFile(f fs.File) (fsapi.Stat_t, syscall.Errno) {
+func statFile(f fs.File) (sys.Stat_t, syscall.Errno) {
 	return defaultStatFile(f)
 }
 
-func inoFromFileInfo(_ string, t fs.FileInfo) (ino fsapi.Ino, err syscall.Errno) {
+func inoFromFileInfo(_ string, t fs.FileInfo) (ino sys.Ino, err syscall.Errno) {
```

Comment:
Should this function also be attempting to cast to `sys.Stat_t`? Just trying to understand the distinction.
Infact can all these `stat_*` implementations be single sourced if they're going through `sys.Stat_t`?

Reply:
this is a little more complicated because we need to get the ino, and we have a path to get it in windows (pun intended)

That said, later I'll try to see if we can unwind and still do that. Good to revisit it!

Reply:
I see what you meant originally and you are right!